checker/src/org/checkerframework/checker/index/samelen/SameLenAnnotatedTypeFactory.java

```diff
@@ -73,6 +77,42 @@ public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
         return new SameLenQualifierHierarchy(factory);
     }
 
+    @Override
+    public AnnotatedTypeMirror getAnnotatedTypeLhs(Tree tree) {
+        AnnotatedTypeMirror atm = super.getAnnotatedTypeLhs(tree);
+        if (tree.getKind() == Tree.Kind.VARIABLE) {
+            String varName = null;
+            Element elt = TreeUtils.elementFromDeclaration((VariableTree) tree);
+            if (elt.getKind() == ElementKind.LOCAL_VARIABLE
```

Comment:
That code doesn't result in a `Receiver`, though - it results in a `String` (which is what I need here; I only get a `Receiver` in the other cases because that's the most straightforward way to get the name.